src/analysis/auto_function.d

```diff
@@ -93,11 +94,63 @@ public:
 	override void visit(const(PrimaryExpression) exp)
 	{
 		exp.accept(this);
-		import std.algorithm.searching : find;
-		import std.range : empty;
-		if (_returns.length && _mixinDepth && !exp.primary.text.find("return").empty)
-			_returns[$-1] = true;
+
+		import std.algorithm.searching : canFind;
+
+		if (_returns.length && _mixinDepth)
+        {
+            if (findReturnInLiteral(exp.primary.text))
+			    _returns[$-1] = true;
+            else if (exp.identifierOrTemplateInstance &&
+                _literalWithReturn.canFind(exp.identifierOrTemplateInstance.identifier.text))
+                    _returns[$-1] = true;
+        }
 	}
+
+    bool findReturnInLiteral(const(string) value)
+    {
+        import std.algorithm.searching : find;
+        import std.range : empty;
+
+        return value == "return" || !value.find("return ").empty;
```

Comment:
Despite of the complexity added for the last case, this will also be better on the previous code. I think it should handle full strings with a return , e.g `"expression; expression; return 0"` but also partially concatenated strings, e.g `return` ~ many other expressions....